app/src/main/java/com/github/se/eventradar/viewmodel/HostedEventsViewModel.kt

```diff
@@ -0,0 +1,66 @@
+package com.github.se.eventradar.viewmodel
+
+import android.util.Log
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.github.se.eventradar.model.Resource
+import com.github.se.eventradar.model.event.EventList
+import com.github.se.eventradar.model.repository.event.IEventRepository
+import com.github.se.eventradar.model.repository.user.IUserRepository
+import dagger.hilt.android.lifecycle.HiltViewModel
+import javax.inject.Inject
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.launch
+
+@HiltViewModel
+class HostedEventsViewModel
+@Inject
+constructor(
+    private val eventRepository: IEventRepository,
+    private val userRepository: IUserRepository
+) : ViewModel() {
+  private val _uiState = MutableStateFlow(HostedEventsUiState())
+  val uiState: StateFlow<HostedEventsUiState> = _uiState
+
+  fun getHostedEvents(uid: String) {
+    viewModelScope.launch {
+      when (val userResponse = userRepository.getUser(uid)) {
+        is Resource.Success -> {
+          val user = userResponse.data!!
+          val eventsHostList = user.eventsHostList
+          if (eventsHostList.isNotEmpty()) {
+            when (val events = eventRepository.getEventsByIds(eventsHostList)) {
+              is Resource.Success -> {
+                _uiState.value =
+                    _uiState.value.copy(
+                        eventList =
+                            EventList(
+                                events.data, events.data, _uiState.value.eventList.selectedEvent))
+              }
+              is Resource.Failure -> {
+                Log.d("HostedEventsViewModel", "Error getting hosted events for $uid")
+                _uiState.value =
+                    _uiState.value.copy(eventList = EventList(emptyList(), emptyList(), null))
+              }
+            }
+          } else {
+            _uiState.value =
```

Comment:
Is this necessary? I think the `eventList` variable is already initialized with these values.

Reply:
I think that it is needed because in the situation where we delete a hosted event from the database, then we call getHostedEvents(uid), if it previously had a non-empty EventList value, it could be recomposed with this even though the event is no longer in the database as it takes the previous EventList value (which will not be updated without this else part), because the empty eventList is only initialized. I'm assuming this because the viewModel state is kept across recompositions (but I'm not sure if this is right)